distribution/integration/tests-integration/src/test/java/org/wso2/carbon/integration/tests/carbontools/CarbonServerBasicOperationTestCase.java

```diff
@@ -0,0 +1,219 @@
+            throw new SkipException("Feature --start is not available for windows");
+        } else {
+            cmdArrayToStart = new String[]
+                    {"sh", "wso2server.sh", "--start", "-DportOffset=" + portOffset};
+            process = CarbonCommandToolsUtil.runScript(carbonHome + "/bin", cmdArrayToStart);
+        }
+        boolean startupStatus = CarbonCommandToolsUtil.isServerStartedUp(context, portOffset);
+        // Waiting until start the server
+
+        Field field = process.getClass().getDeclaredField("pid");
+        field.setAccessible(true);
+        processId = field.get(process).toString();
+        log.info("process id for carbon server with offset 1 : " + processId);
+
+        assertTrue(startupStatus, "Unsuccessful login");
+    }
+
+    @Test(groups = {"wso2.as"}, description = "Server start test", dependsOnMethods = {"testStartCommand"})
+    public void testDumpCommandOnLinux() throws Exception {
+        String[] cmdArray;
+        Process processDump = null;
+        try {
+            if (CarbonCommandToolsUtil.isCurrentOSWindows()) {
+                throw new SkipException("Feature --start is not available for windows");
+                // Since we are skipping --start feature it won
+            } else {
+                cmdArray = new String[]
+                        {"sh", "carbondump.sh", "-carbonHome", carbonHome, "-pid", processId};
+                processDump = CarbonCommandToolsUtil.runScript(carbonHome + "/bin", cmdArray);
+            }
+            assertTrue(isFoundDumpFile(carbonHome), "Couldn't find the dump file");
+        } finally {
+            processDump.destroy();
+        }
+
+    }
+
+    @Test(groups = {"wso2.as"}, description = "Server restart test",
+            dependsOnMethods = {"testDumpCommandOnLinux"})
+    public void testRestartCommand() throws Exception {
+        String[] cmdArrayToReStart;
+        if (CarbonCommandToolsUtil.isCurrentOSWindows()) {
+            throw new SkipException("Feature --restart is not available for windows");
+        } else {
+            cmdArrayToReStart = new String[]
+                    {"sh", "wso2server.sh", "--restart", "-DportOffset=" + portOffset};
+            CarbonCommandToolsUtil.runScript(carbonHome + "/bin", cmdArrayToReStart);
+        }
+
+        boolean isServerDown = CarbonCommandToolsUtil.isServerDown(context, portOffset);
+        assertTrue(isServerDown, "Shutting down the server failed");
+
+        boolean isServerUp = CarbonCommandToolsUtil.isServerStartedUp(context, portOffset);
+
+        assertTrue(isServerUp, "Unsuccessful login");
+    }
+
+    @Test(groups = {"wso2.as"}, description = "Server stop test",
+            dependsOnMethods = {"testRestartCommand"})
+    public void testStopCommand() throws Exception {
+        String[] cmdArray;
+        boolean startupStatus = false;
+        try {
+            if (CarbonCommandToolsUtil.isCurrentOSWindows()) {
```

Comment:
we need to change this set to return operating system from CarbonCommandToolsUtil 
if(CarbonCommandToolsUtil.getOperatingSyatem().equals(OperatinSystam.WINDOWS)